test/tokens/ERC1155/presets/ERC1155Lootbox.t.sol

```diff
@@ -0,0 +1,280 @@
+    }
+
+    function testFactoryDetermineAddress(
+        address _proxyOwner,
+        address tokenOwner,
+        string memory name,
+        string memory symbol,
+        string memory baseURI,
+        string memory contractURI,
+        address royaltyReceiver,
+        uint96 royaltyFeeNumerator
+    ) public {
+        vm.assume(_proxyOwner != address(0));
+        vm.assume(tokenOwner != address(0));
+        vm.assume(royaltyReceiver != address(0));
+        royaltyFeeNumerator = uint96(bound(royaltyFeeNumerator, 0, 10_000));
+        ERC1155LootboxFactory factory = new ERC1155LootboxFactory(address(this));
+        address deployedAddr =
+            factory.deploy(_proxyOwner, tokenOwner, name, baseURI, contractURI, royaltyReceiver, royaltyFeeNumerator);
+        address predictedAddr = factory.determineAddress(
+            _proxyOwner, tokenOwner, name, baseURI, contractURI, royaltyReceiver, royaltyFeeNumerator
+        );
+        assertEq(deployedAddr, predictedAddr);
+    }
+
+    function testCommitWithBalance(address user) public {
+        _commit(user);
+        vm.assertEq(lootbox.balanceOf(user, 1), 0);
+    }
+
+    function testCommitNoBalance(address user) public {
+        vm.prank(user);
+        vm.expectRevert(NoBalance.selector);
+        lootbox.commit();
+    }
+
+    function testRefundNoCommit(address user) public {
+        vm.expectRevert(NoCommit.selector);
+        lootbox.refundBox(user);
+    }
+
+    function testRefundPendingReveal(address user) public {
+        _commit(user);
+
+        vm.roll(block.number + 255);
+        vm.expectRevert(PendingReveal.selector);
+        lootbox.refundBox(user);
+    }
+
+    function testRefundExpiredCommit(address user) public {
+        _commit(user);
+
+        vm.roll(block.number + 300);
+        lootbox.refundBox(user);
+        vm.assertEq(lootbox.balanceOf(user, 1), 1);
+    }
+
+    function testGetRevealIdNoCommit(address user) public {
+        vm.expectRevert(NoCommit.selector);
+        lootbox.getRevealId(user);
+    }
+
+    function testGetRevealIdInvalidCommit(address user) public {
+        _commit(user);
+
+        vm.roll(block.number + 300);
+        vm.expectRevert(InvalidCommit.selector);
+        lootbox.getRevealId(user);
+    }
+
+    function testGetRevealIdSuccess(address user) public {
+        vm.assertLt(_getRevealId(user), lootbox.boxSupply());
+    }
+
+    function testRevealSuccess(address user) public {
+        uint256 revealIdx = _getRevealId(user);
+
+        (, bytes32[] memory proof) = TestHelper.getMerklePartsBoxes(boxContents, revealIdx);
+
+        IERC1155LootboxFunctions.BoxContent memory boxContent = boxContents[revealIdx];
+
+        lootbox.reveal(user, boxContent, proof);
+
+        for (uint256 i = 0; i < boxContent.tokenAddresses.length; i++) {
+            vm.assertEq(token.balanceOf(user, boxContent.tokenIds[i]), boxContent.amounts[i]);
+        }
+    }
+
```

Comment:
Can we add a reentry attack test on `reveal`? Where `reveal` mints tokens to a contract that calls `reveal` again